pkg/consuldp/bootstrap.go

```diff
@@ -54,6 +54,14 @@ func (cdp *ConsulDataplane) bootstrapConfig(ctx context.Context) (*bootstrap.Boo
 		return nil, nil, fmt.Errorf("failed to get envoy bootstrap params: %w", err)
 	}
 
+	// store the final resolved service for others to use.
+	cdp.resolvedProxyConfig = ProxyConfig{
+		NodeName:  rsp.NodeName,
+		ProxyID:   cdp.cfg.Proxy.ProxyID,
+		Namespace: rsp.Namespace,
+		Partition: rsp.Partition,
+	}
```

Comment:
Would you expect that the node, namespace + partition would sometimes differ between `rsp` and `cdp.cfg.Proxy`? I ask because I was curious why we can't just use `cfg.cfg.Proxy` directly vs. hanging an additional `ProxyConfig` on the struct

Reply:
Matt might know differently, but I assumed it was that the serviceaccount/Token used to make the bootstrapping could imply a namespace/partition that wasn't specified using the config flags.